### ISSUE TITLE
Idea/RFC: After Running PNG Toolchain, Process IDAT via Zip ToolChain

### ISSUE THREAD

**Issue by TPS** [javiergutierrezchamorro/nikkhokkho#6]
@javiergutierrezchamorro I was browsing @glennrp's repos recently, & I ran across the [pngzop readme](https://github.com/glennrp/pmt/blob/pngzop/pngzop-1.0.2-README.txt).  IIUC, in a nutshell, [pngzop](https://github.com/glennrp/pmt/tree/pngzop) is _([was?](https://github.com/glennrp/pmt/compare/pngzop...pngcrush?expand=0))_ his effort to integrate [zopfli compression](https://github.com/google/zopfli) directly into his [pngcrush](https://github.com/glennrp/pmt/tree/pngcrush), bypassing [zopflipng](https://github.com/google/zopfli/tree/master/src/zopflipng)'s own filter heuristics.  In that method,  pngzop would eventually dissect the PNG, pull the IDAT out, & process via [zopfli](https://github.com/google/zopfli/tree/master/src/zopfli) (_not_ zopflipng), & then reassemble.

Since [FileOptmizer](https://github.com/svn2github/nikkhokkho/tree/master/FileOptimizer) has a _far_ better zip toolchain than just zopfli, maybe it'd be worth integrating the pngzop dissection-&-reassembly tools (&/or any others, at your preference) into a similar process into the png toolchain.

Thoughts/comments/interest?

**Comment by javiergutierrezchamorro** [comment]
I will try to do a Win32 and Win64 builds of 1.8.1, and try them.

**Comment by TPS** [comment]
>I will try to do a Win32 and Win64 builds

@javiergutierrezchamorro I was searching for Windows builds of the various `pngzop`-specific tools (in the source tree as `pngzop_*.c`), & none seem to exist.  Did you have any luck?  If so & you'd like to post them someplace, I'd like to try them out manually to work out if it's worth converting the [`pngzop` `sh` script](https://github.com/glennrp/pmt/blob/pngzop/pngzop) to a `cmd`/batch, or maybe write my own.

**Comment by javiergutierrezchamorro** [comment]
Not yet. I have been quite busy with my real job lately, while FileOptimizer efforts have gone on the side of optional lossy recompression (JPEG and GIF), as well as adding more formats, and improving optimization ratio.

**Comment by TPS** [comment]
@xdevs23 Do you have Windows binaries?

**Comment by TPS** [comment]
@javiergutierrezchamorro: @MrKrzYch00 of [Zopfli KrzYmod](https://github.com/MrKrzYch00/zopfli) fame has made available `krzydefc` ([README](https://github.com/MrKrzYch00/krzydefc), [binary](https://encode.ru/threads/2176-Zopfli-amp-ZopfliPNG-KrzYmod)) to convert ↔ {GZ, ZIP, ZLIB, PNG}, which is _specifically_ designed to effect this kind of toolchain.  Would you interested in taking a look?

_(Pointed out in https://github.com/sayurin/optipng-zopfli/issues/2#issuecomment-368951693 onwards.)_

**Comment by javiergutierrezchamorro** [comment]
Thanks @TPS 
According to what I saw it seems to be always compressing files inside a ZIP or GZ container.
So what would be the idea? Uncompress original ZIP/GZ, and then compress with Zopfli KrzYmod?

**Comment by TPS** [comment]
Nono, after running PNG through typical FO toolchain, convert to ZIP/GZ (whichever has better tools), run through FO as such, & then convert back.  Or whatever permutation of that is optimal.  `krxyfefc` isn't _specific_ to only @MrKrzYch00's tools, I believe.

**Comment by TPS** [comment]
Similarly, after running **A**PNG through its dedicated toolchain, use [**A**PNG DisAssembler](https://sourceforge.net/projects/apngdis/), run through PNG → ZIP toolchains ([as detailed above](https://github.com/javiergutierrezchamorro/nikkhokkho/issues/6#issuecomment-373777742)), & then use [**A**PNG Assembler](https://sourceforge.net/projects/apngasm/).  Or vice versa, &c, in optimized order.

**Comment by javiergutierrezchamorro** [comment]
> Similarly, after running **A**PNG through its dedicated toolchain (mostly [**A**PNG Optimizer](https://sourceforge.net/projects/apng/files/APNG_Optimizer/)?), use [**A**PNG DisAssembler](https://sourceforge.net/projects/apngdis/), run through PNG → ZIP toolchains ([as detailed above](https://github.com/javiergutierrezchamorro/nikkhokkho/issues/6#issuecomment-373777742)), & then use [**A**PNG Assembler](https://sourceforge.net/projects/apngasm/). Or vice versa, &c, in optimized order.

Not as easy because once APNG is disassembled, you are loosing the information of duration of each frame, and thus you will not be able to reconstruct again with the assembler.

**Comment by TPS** [comment]
Of course, 1 isn't (yet) deleting the original, right?  So, that'd be used as the template to *re*assemble correctly.

**Comment by javiergutierrezchamorro** [comment]
> Of course, 1 isn't (yet) deleting the original, right? So, that'd be used as the template to _re_assemble correctly.

Sure. But this will require implementing additional code on FO just to read and keep track of all frame information.

**Comment by TPS** [comment]
[TweakPNG Command Line](https://web.archive.org/web/20090530154000/http://www.trenttompkins.com:80/Downloads/TweakPNG-Command-Line.html) _might_ be able to help w/ that, though (clearly) no longer maintained.  I'll ask whether its command-line bits can be supported in mainline [TweakPNG](https://github.com/jsummers/tweakpng).
